File: app/lib/features/auth/domain/app_user.dart
/// The authenticated user (backend `PublicUser`). Identity comes from SBC.
class AppUser {
  const AppUser({
    required this.id,
    required this.sbcUserId,
    this.name,
    this.email,
    this.phoneNumber,
    this.country,
    this.avatarUrl,
    this.subscriptionTypes = const [],
    this.isActivated = false,
    this.role = 'USER',
  });

  factory AppUser.fromJson(Map<String, dynamic> json) => AppUser(
        id: (json['id'] ?? '').toString(),
        sbcUserId: (json['sbcUserId'] ?? '').toString(),
        name: json['name'] as String?,
        email: json['email'] as String?,
        phoneNumber: json['phoneNumber'] as String?,
        country: json['country'] as String?,
        avatarUrl: json['avatarUrl'] as String?,
        subscriptionTypes: (json['subscriptionTypes'] as List<dynamic>? ?? [])
            .map((e) => e.toString())
            .toList(),
        isActivated: json['isActivated'] as bool? ?? false,
        role: (json['role'] ?? 'USER').toString(),
      );

  final String id;
  final String sbcUserId;
  final String? name;
  final String? email;
  final String? phoneNumber;
  final String? country;
  final String? avatarUrl;
  final List<String> subscriptionTypes;
  final bool isActivated;
  final String role;

  bool get hasActiveSubscription => subscriptionTypes.isNotEmpty;
  String get displayName => (name == null || name!.isEmpty) ? 'Membre SBC' : name!;
}

File: ios/SBCContacts/Features/Auth/AppUser.swift
import Foundation

/// The authenticated user (backend `PublicUser`). Identity comes from SBC.
struct AppUser: Sendable, Equatable {
    var id: String
    var sbcUserId: String
    var name: String?
    var email: String?
    var phoneNumber: String?
    var country: String?
    var avatarUrl: String?
    var subscriptionTypes: [String] = []
    var isActivated = false
    var role = "USER"

    var hasActiveSubscription: Bool { !subscriptionTypes.isEmpty }
    var displayName: String { (name?.isEmpty ?? true) ? "Membre SBC" : name! }
}

extension AppUser: Decodable {
    private enum CodingKeys: String, CodingKey {
        case id, sbcUserId, name, email, phoneNumber, country, avatarUrl, subscriptionTypes, isActivated, role
    }

    init(from decoder: Decoder) throws {
        let c = try decoder.container(keyedBy: CodingKeys.self)
        id = c.string(.id) ?? ""
        sbcUserId = c.string(.sbcUserId) ?? ""
        name = c.string(.name)
        email = c.string(.email)
        phoneNumber = c.string(.phoneNumber)
        country = c.string(.country)
        avatarUrl = c.string(.avatarUrl)
        subscriptionTypes = c.strings(.subscriptionTypes)
        isActivated = c.bool(.isActivated) ?? false
        role = c.string(.role) ?? "USER"
    }
}

/// `POST /auth/sso-callback` response.
struct SSOSession: Decodable, Sendable {
    struct Tokens: Decodable, Sendable {
        let accessToken: String
        let refreshToken: String
    }

    let tokens: Tokens
    let user: AppUser
}
